Src/Plugins/ValidatorBase.cs

```diff
@@ -62,26 +63,38 @@ protected ValidatorBase()
                                            ref string fingerprint,
                                            ref string message)
         {
-            string state =
-                validator.IsValidStaticHelper(ref matchedPattern,
-                                              ref groups,
-                                              ref failureLevel,
-                                              ref fingerprint,
-                                              ref message);
-
-            if (state != nameof(ValidationState.NoMatch))
+            validator.MatchCleanup(ref matchedPattern,
+                                    ref groups,
+                                    ref failureLevel,
+                                    ref fingerprint,
+                                    ref message);
+
+            return PerformValidationAndCheckCache(validator, ref matchedPattern, ref groups, ref failureLevel, ref fingerprint, ref message);
+        }
+
+        protected static string PerformValidationAndCheckCache(ValidatorBase validator, ref string matchedPattern, ref Dictionary<string, string> groups, ref string failureLevel, ref string fingerprint, ref string message)
+        {
+            string state = validator.IsValidStaticHelper(ref matchedPattern,
+                                                                      ref groups,
+                                                                      ref failureLevel,
+                                                                      ref fingerprint,
+                                                                      ref message);
```

Comment:
this alignment isn't right, style NIT.

Reply:
Was expecting BuildAndTest to fix it.  My bad.

---
In reply to: [575502466](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575502466) [](ancestors = 575502466)